src/main/java/org/assertj/core/internal/Strings.java

```diff
@@ -878,7 +878,23 @@ private void checkRegexIsNotNull(CharSequence regex) {
   public void assertMatches(AssertionInfo info, CharSequence actual, Pattern pattern) {
     checkIsNotNull(pattern);
     assertNotNull(info, actual);
-    if (!pattern.matcher(actual).matches()) throw failures.failure(info, shouldMatch(actual, pattern.pattern()));
+    assertMatches(info, actual, pattern.matcher(actual));
+  }
+
+  /**
+   * Verifies that the given {@code CharSequence} matches the given regular expression.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code CharSequence}.
+   * @param matcher the matcher to check for matching.
+   * @throws NullPointerException if the given pattern is {@code null}.
+   * @throws AssertionError if the given {@code CharSequence} is {@code null}.
+   * @throws AssertionError if the given {@code CharSequence} does not match the given regular expression.
+   */
+  public void assertMatches(AssertionInfo info, CharSequence actual, Matcher matcher) {
```

Comment:
add unit tests for this method, look at `Strings_assertIsBase64_Test` for a good example writing tests in AssertJ